### ISSUE TITLE
Make an ES 2.1.1 compatible version

### ISSUE THREAD

**Issue by danielschonfeld** [fabric8io/elasticsearch-cloud-kubernetes#24]
Requesting a version that's installable on ES 2.1.1.  Not sure if any 'real' changes are required for this to become a reality.

**Comment by anil-g** [comment]
Hello Daniel,

Given that 2.1.1 is just a minor revision, I suppose the 2.1.0 version
should be compatible. Did you try 2.1.0?

plugin install io.fabric8/elasticsearch-cloud-kubernetes/2.1.0

**Comment by danielschonfeld** [comment]
@anil-g yes i did and there's an error generated by the plugin script that says it's not compatible.  That's what prompted me to open this issue.

**Comment by jimmidyson** [comment]
I remember reading that it must be compiled with the exact target version... that sucks tbh so I hope there's a nicer way to do that - appreciate help if anyone knows?

**Comment by kujenga** [comment]
I'm seeing the same issue, output below. In the interim, would it be possible to just deploy the 2.1.1 version?

```
Step 1 : FROM elasticsearch:2.1.1
 ---> a05cc7ed3f32
Step 2 : RUN /usr/share/elasticsearch/bin/plugin install io.fabric8/elasticsearch-cloud-kubernetes/2.1.0 --verbose
 ---> Running in 397faad23aa6
-> Installing io.fabric8/elasticsearch-cloud-kubernetes/2.1.0...
Trying https://download.elastic.co/io.fabric8/elasticsearch-cloud-kubernetes/elasticsearch-cloud-kubernetes-2.1.0.zip ...
Failed: FileNotFoundException[https://download.elastic.co/io.fabric8/elasticsearch-cloud-kubernetes/elasticsearch-cloud-kubernetes-2.1.0.zip]; nested: FileNotFoundException[https://download.elastic.co/io.fabric8/elasticsearch-cloud-kubernetes/elasticsearch-cloud-kubernetes-2.1.0.zip]; 
Trying https://search.maven.org/remotecontent?filepath=io/fabric8/elasticsearch-cloud-kubernetes/2.1.0/elasticsearch-cloud-kubernetes-2.1.0.zip ...
Downloading .................................................................................................................................................................................................................................................................................................................................DONE
Verifying https://search.maven.org/remotecontent?filepath=io/fabric8/elasticsearch-cloud-kubernetes/2.1.0/elasticsearch-cloud-kubernetes-2.1.0.zip checksums if available ...
Downloading .DONE
ERROR: Plugin [cloud-kubernetes] is incompatible with Elasticsearch [2.1.1]. Was designed for version [2.1.0]
The command '/bin/sh -c /usr/share/elasticsearch/bin/plugin install io.fabric8/elasticsearch-cloud-kubernetes/2.1.0 --verbose' returned a non-zero code: 64
```

**Comment by kujenga** [comment]
Perhaps you could try specifying the version of this plugin to be `2.1` rather than `2.1.0`, that might get around the specificity issue.

**Comment by danielschonfeld** [comment]
@kujenga results:

```
-> Installing io.fabric8/elasticsearch-cloud-kubernetes/2.1...
Trying https://download.elastic.co/io.fabric8/elasticsearch-cloud-kubernetes/elasticsearch-cloud-kubernetes-2.1.zip ...
Trying https://search.maven.org/remotecontent?filepath=io/fabric8/elasticsearch-cloud-kubernetes/2.1/elasticsearch-cloud-kubernetes-2.1.zip ...
Trying https://oss.sonatype.org/service/local/repositories/releases/content/io/fabric8/elasticsearch-cloud-kubernetes/2.1/elasticsearch-cloud-kubernetes-2.1.zip ...
Trying https://github.com/io.fabric8/elasticsearch-cloud-kubernetes/archive/2.1.zip ...
Trying https://github.com/io.fabric8/elasticsearch-cloud-kubernetes/archive/master.zip ...
ERROR: failed to download out of all possible locations..., use --verbose to get detailed information
The command '/bin/sh -c /usr/share/elasticsearch/bin/plugin install io.fabric8/elasticsearch-cloud-kubernetes/2.1' returned a non-zero code: 74
```

**Comment by kujenga** [comment]
Sorry, @danielschonfeld that was meant to be a suggestion for the project maintainers, I tried it as well 😊

**Comment by jimmidyson** [comment]
I think @kujenga meant in the plugin.properties in the zip.

**Comment by jimmidyson** [comment]
I've tried this but it doesn't work. Seems like you need to match exact server version. Will publish a compatible version later today.

**Comment by pires** [comment]
@jimmidyson later _today_ will support 2.2.0 as well, right? :joy:

**Comment by pires** [comment]
2.2.1 is about to be released. Is this still needed?

**Comment by jimmidyson** [comment]
Already released a 2.1.1 version (http://repo1.maven.org/maven2/io/fabric8/elasticsearch-cloud-kubernetes/2.1.1/) anyway so closing.
